src/Umbraco.Web.UI.Client/src/views/dashboard/developer/redirecturls.controller.js

```diff
@@ -0,0 +1,144 @@
+(function() {
+    "use strict";
+
+    function RedirectUrlsController($scope, redirectUrlsResource, notificationsService, $q) {
+        //...todo
+        //search by url or url part
+        //search by domain
+        //display domain in dashboard results?
+
+        //used to cancel any request in progress if another one needs to take it's place
+        var vm = this;
+        var canceler = null;
+
+        vm.dashboard = {
+            searchTerm: "",
+            loading: false,
+            UrlTrackerDisabled: false
+        };
+
+        vm.pagination = {
+            pageIndex: 0,
+            pageNumber: 1,
+            totalPages: 1,
+            pageSize: 20
+        };
+
+        vm.goToPage = goToPage;
+        vm.search = search;
+        vm.removeRedirect = removeRedirect;
+        vm.disableUrlTracker = disableUrlTracker;
+        vm.enableUrlTracker = enableUrlTracker;
+        vm.filter = filter;
+
+        function activate() {
+            vm.search();
+        }
+
+        function goToPage(pageNumber) {
+            vm.pagination.pageIndex = pageNumber - 1;
+            vm.pagination.pageNumber = pageNumber;
+            vm.search();
+        }
+
+        function search() {
+
+            vm.dashboard.loading = true;
+
+            var searchTerm = vm.dashboard.searchTerm;
+            if (searchTerm === undefined) {
+                searchTerm = "";
+            }
+
+            redirectUrlsResource.searchRedirectUrls(searchTerm, vm.pagination.pageIndex, vm.pagination.pageSize).then(function(response) {
+
+                vm.redirectUrls = response.SearchResults;
+
+                // update pagination
+                vm.pagination.pageIndex = response.CurrentPage;
+                vm.pagination.pageNumber = response.CurrentPage + 1;
+                vm.pagination.totalPages = response.PageCount;
+
+                // Set enable/disable state for url tracker
+                vm.dashboard.UrlTrackerDisabled = response.UrlTrackerDisabled;
+
+                angular.forEach(vm.redirectUrls, function(redirect) {
+                    redirectUrlsResource.getPublishedUrl(redirect.ContentId).then(function(response) {
```

Comment:
Not sure why this is coded like this - this causes N+1 requests. This means that if there are 100s of redirect rules loaded per page we're going to make 100 requests , this needs to be fixed, there's no reason why this data isn't included in the initial search result. ... unless I'm missing something?